packages/starlight/__e2e__/test-utils.ts

```diff
@@ -25,6 +25,8 @@ export function testFactory(fixturePath: string) {
 		} = {}
 	): Promise<Server> {
 		const { mode } = options;
+		// Reset NODE_ENV so it can be re-set by `dev()` or `build()`.
+		delete process.env.NODE_ENV;
```

Comment:
Maybe this should also contains a todo comment to remove this once fixed? Would hate to miss an issue in the long run because we're doing this in an E2E test but that's not something users do.

Reply:
Considering that Matt created an amazing PR fixing it, I guess this comment is irrelevant now 😄

Reply:
Astro moves fast!